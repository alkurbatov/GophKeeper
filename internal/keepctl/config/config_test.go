package config_test

import (
	"os"
	"strings"
	"testing"

	"github.com/alkurbatov/goph-keeper/internal/keepctl/config"
	"github.com/alkurbatov/goph-keeper/internal/libraries/gophtest"
	"github.com/gkampitakis/go-snaps/snaps"
)

func unsetGophEnv() {
	for _, env := range os.Environ() {
		if strings.HasPrefix(env, "GOPH_") {
			pair := strings.SplitN(env, "=", 2)

			os.Unsetenv(pair[0])
		}
	}
}

func TestDefaultConfig(t *testing.T) {
	unsetGophEnv()

	sat := config.New()

	snaps.MatchSnapshot(t, sat.String())
}

func TestConfigFromEnv(t *testing.T) {
	os.Setenv("GOPH_USERNAME", gophtest.Username)
	os.Setenv("GOPH_PASSWORD", string(gophtest.Password))
	os.Setenv("GOPH_ADDRESS", "192.168.0.10:8080")
	os.Setenv("GOPH_CA_PATH", "/etc/ssl/root.crt")
	os.Setenv("GOPH_VERBOSE", "1")

	t.Cleanup(unsetGophEnv)

	sat := config.New()

	snaps.MatchSnapshot(t, sat.String())
}
