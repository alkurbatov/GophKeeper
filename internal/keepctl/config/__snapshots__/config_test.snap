
[TestDefaultConfig - 1]
Configuration:
        Username: 
        Password: 
        Keeper address: 127.0.0.1:50051
        Certificate authority path: 
        Verbose: false
---

[TestConfigFromEnv - 1]
Configuration:
        Username: admin
        Password: ******
        Keeper address: 192.168.0.10:8080
        Certificate authority path: /etc/ssl/root.crt
        Verbose: true
---
